core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java

```diff
@@ -34,8 +34,7 @@ public static File createDirectory(String path, String name, File file, File sub
         File directory;
         String fileName = file.getPath();
         String submissionRootPath = submissionRoot.getPath();
-        int lastDirectoryIndex = findRootDirIndex(name, submissionRootPath);
-        fileName = fileName.substring(lastDirectoryIndex).replaceFirst(name, "");
+        fileName = fileName.substring(submissionRootPath.length());
```

Comment:
If you only want the name, you can fo `file.getName()` instead.

Reply:
Actually, it is with a separator, for example, \TerrainType.java or /TerrainType.java.

Reply:
Then I would suggest `String fileName = File.separator() + file.getName();`. This makes it clearer what is actually required there.